### PR TITLE
use cookies to remember where you left from Explore page

### PR DESCRIPTION
**Pull request by 66eli77**
If you somewhere inside the Explore page, when you left the page then come back, kolibri will redirect you to where you left in Explore.

### CODE REVIEW

File: kolibri/plugins/learn/assets/src/actions.js [hunk 1]
```diff
@@ -159,12 +159,30 @@ function redirectToExploreChannel(store) {
...
+            },
+          });
+        } else {
+          router.go({
+            name: constants.PageNames.EXPLORE_CHANNEL,
+            params: {
+              channel_id: currentChannelId,
+            },
+          });
+        }
```
Comment (indirectlylit) [inline kolibri/plugins/learn/assets/src/actions.js:185]: This isn't an the appropriate location for this logic... a method called `_getCurrentChannelId` shouldn't have a bunch of cookie-based routing in it.

Comment (66eli77) [reply kolibri/plugins/learn/assets/src/actions.js:185]: the reason being?

Comment (rtibbles) [reply kolibri/plugins/learn/assets/src/actions.js:185]: @indirectlylit Note that this is not inside the _getCurrentChannelId function, it is chained from the returned Promise from that function.

**Comment by rtibbles** [comment]
I don't think this should be merged, and I think the 'issue' it is fixing is expected behaviour.

We are not creating an OS, @66eli77!

**Comment by 66eli77** [comment]
@rtibbles  imagine you finally find the topic in Explore at cancer>RV>meth>cooking>oblivion, now you want to checkout Learn page or Management page and come back to oblivion. You'd have to click through the path again.

**Comment by rtibbles** [comment]
You can use the back button.

Well, you can after this PR is fixed so that we no longer break the back button: https://github.com/learningequality/kolibri/pull/490

**Comment by 66eli77** [comment]
on mobile webview, there's no back button

**Comment by rtibbles** [comment]
As in, when we have an Android app with an embedded webview in it? Why wouldn't our app have a back button in that case?

**Comment by rtibbles** [comment]
We can even intercept the overall Android 'back' so as to save screen space: http://stackoverflow.com/questions/16152491/enable-back-button-in-webview

**Comment by 66eli77** [comment]
say you navigate to management page and did something that changed the url, you then click the back btn couple times to go back to Explore, will it revert what you did in management?

**Comment by rtibbles** [comment]
As in, would it uncreate a user? We have not been using URL navigation for any actions, so I can't see how.

**Comment by 66eli77** [comment]
also, mobile app should not solely reply on the universal back btn for in app navigation, I think the same also apply to website, the site should contain all  necessary nav elements

**Comment by rtibbles** [comment]
So you want to add a 'back' button that replicates browser back into our page?

**Comment by jamalex** [comment]
> So you want to add a 'back' button that replicates browser back into our page?

So 1990's!

**Comment by 66eli77** [comment]
I don't think we should rely on the browser back btn either. The browser back btn is designed for real website, real website that has external links and meant to be able to jump between different domains. But Kolibri is more like a application. We don't encourage user jump in and out of the app. Therefore, it should not rely on the browser back btn.

**Comment by rtibbles** [comment]
It's a web app, it runs in the browser, it follows web standards for many things, including using the back button to retrace navigation steps.

**Comment by jamalex** [comment]
Kolibri is a real website, used in a web browser, by people we want to teach how to use other real websites. It will also, in practice, be used in many contexts embedded in other "websites" -- e.g. RACHEL, and @benjaoming's resource bundle.

We're not trying to reinvent the web experience, we're just trying to make it available to people who otherwise would have access issues.

**Comment by jamalex** [comment]
> Explore at cancer>RV>meth>cooking>oblivion, now you want to checkout Learn page or Management page and come back to oblivion.

First of all, oblivion already blew up. :boom: 

Secondly, if you did some stuff in that topic and then left, it will (very soon, once @ralphiee22's new stuff is in) show up under your "Resume" section in "Learn".

**Pull request closed without merging**